core/src/main/scala/org/apache/gearpump/cluster/Application.scala

```diff
@@ -27,15 +27,16 @@ import org.apache.gearpump.jarstore.JarFileContainer
  * This contains all information to run an application
  *
  * @param name: The name of this application
- * @param conf: user configuration.
+ * @param userConfig: user configuration.
  * @param appMaster: The class name of AppMaster Actor
+ * @param appMasterClusterConfig: The akka config source used to create the appmaster actor system
  *    The AppMaster must have a constructor this(appContext : AppMasterContext, app : Application)
  */
 
-final class Application(val name : String, val appMaster : String, val conf: UserConfig) extends Serializable
+final class Application(val name : String, val appMaster : String, val userConfig: UserConfig, val appMasterClusterConfig: ClusterConfigSource = null) extends Serializable
 
```

Comment:
appMasterClusterConfig is used to initialize the akka system used by appmaster.

Will add more comments.